server/src/test/java/io/crate/execution/engine/collect/files/FileReadingIteratorTest.java

```diff
@@ -232,6 +234,85 @@ public String readLine() throws IOException {
         tester.verifyResultAndEdgeCaseBehaviour(expectedResult);
     }
 
+    @Test
```

Comment:
Not sure, but maybe worth to describe the reason here a bit more verbose.
```suggestion
    /**
     * Validates a bug which was resulting in duplicate reads of the same line when consecutive retries happen.
     */    
    @Test
```

Reply:
It's not bad to add more explanation, personally I'd prefer a link to the issue instead, but as you like.

Reply:
appending the link as well